### ISSUE TITLE
Add support for COCO style datasets for instance segmentation

### ISSUE THREAD

**Issue by roboserg** [huggingface/transformers#25337]
### Feature request

Create a standard dataset loader capable of taking datasets in the JSON COCO style format and converting them into the Huggingface format. The DatasetDict will be generated with the correct features and configurations, making it suitable for various downstream tasks, such as instance segmentation fine-tuning with the Mask2Former mode from Huggingface hub.

The loader should include a flag that allows users to specify the type of segmentation they want to load, such as "panoptic," "semantic," or "instance." These different segmentation tasks store data differently within the COCO format. It is important to note that COCO segmentation masks can be represented in two ways inside the JSON file - either as a polygon or a bitmask.

### Motivation

COCO style formatted datasets are prevalent in computer vision, especially in segmentation. Adopting them would ease the transition for Deep Learning practitioners as data preparation is often the main hurdle.

### Your contribution

I am versed as to how the COCO format is structured but I am a total newbie to Huggingface.

**Comment by rafaelpadilla** [comment]
Hi @roboserg :) 

I created a COCO dataset for bounding boxes only. Maybe it could be useful to you:
https://huggingface.co/datasets/rafaelpadilla/coco2017

You can find a [COCODataset](https://huggingface.co/datasets/rafaelpadilla/coco2017/blob/main/cocodataset/dataset.py#L14) class, which takes the `loaded_json` dictionary representing the JSON containing COCO's bounding boxes. If you find it useful, you could adapt it for other cases (panoptic and semantic). This way, you can have a dataset class to represent COCO's samples and annotations used in a dataloader.
